src/axom/quest/tests/quest_intersection_shaper.cpp

```diff
@@ -0,0 +1,553 @@
+                                               hi[2] - lo[2],
+                                               false));
+  mesh->SetCurvature(polynomialOrder);
+  dc.SetMeshNodesName("positions");
+  dc.SetMesh(mesh);
+
+  // This mode will make 2 clean materials in the mesh.
+  if(initialMats)
+  {
+    mfem::GridFunction *mata = newGridFunction(mesh);
+    mfem::GridFunction *matb = newGridFunction(mesh);
+    for(int k = 0; k < celldims[2]; k++)
+    {
+      for(int j = 0; j < celldims[1]; j++)
+      {
+        for(int i = 0; i < celldims[0]; i++)
+        {
+          int id = k * celldims[1] * celldims[0] + j * celldims[0] + i;
+          (*mata)(id) = (i < celldims[0] / 2);
+          (*matb)(id) = (i >= celldims[0] / 2);
+        }
+      }
+    }
+    dc.RegisterField("vol_frac_a", mata);
+    dc.RegisterField("vol_frac_b", matb);
+  }
+}
+
+// Save Sidre as VisIt
+void saveVisIt(const std::string &path,
+               const std::string &filename,
+               sidre::MFEMSidreDataCollection &dc)
+{
+  // Wrap mesh and grid functions in a VisItDataCollection and save it.
+  mfem::VisItDataCollection vdc(filename, dc.GetMesh());
+  if(!path.empty()) vdc.SetPrefixPath(path);
+  vdc.SetOwnData(false);
+  vdc.SetFormat(mfem::DataCollection::SERIAL_FORMAT);
+  for(auto it : dc.GetFieldMap())
+  {
+    if(it.first.find("vol_frac_") != std::string::npos)
+    {
+      vdc.RegisterField(it.first, it.second);
+    }
+  }
+  vdc.Save();
+}
+
+// Load VisIt as Sidre
+void loadVisIt(mfem::VisItDataCollection &vdc, sidre::MFEMSidreDataCollection &dc)
+{
+  // Wrap mesh and grid functions in a VisItDataCollection and save it.
+  vdc.SetFormat(mfem::DataCollection::SERIAL_FORMAT);
+  vdc.Load();
+  dc.SetOwnData(false);
+  dc.SetMesh(vdc.GetMesh());
+  for(auto it : vdc.GetFieldMap())
+  {
+    if(it.first.find("vol_frac_") != std::string::npos)
+      dc.RegisterField(it.first, it.second);
+  }
+}
+
+// Turn a MFEMSidreDataCollection's fields into a simple Conduit node so
+// I/O is not so problematic.
+void dcToConduit(sidre::MFEMSidreDataCollection &dc, conduit::Node &n)
+{
+  for(auto it : dc.GetFieldMap())
+  {
+    // Just compare vol_frac_ grid functions.
+    if(it.first.find("vol_frac_") != std::string::npos)
+    {
+      n[it.first].set(it.second->GetData(), it.second->Size());
+    }
+  }
+}
+
+bool compareConduit(const conduit::Node &n1,
+                    const conduit::Node &n2,
+                    double tolerance,
+                    conduit::Node &info)
+{
+  bool same = true;
+  if(n1.dtype().id() == n2.dtype().id() && n1.dtype().is_floating_point())
+  {
+    const auto a1 = n1.as_double_accessor();
+    const auto a2 = n2.as_double_accessor();
+    double maxdiff = 0.;
+    for(int i = 0; i < a1.number_of_elements() && same; i++)
+    {
+      double diff = fabs(a1[i] - a2[i]);
+      maxdiff = std::max(diff, maxdiff);
+      same &= diff <= tolerance;
+      if(!same)
```

Comment:
Good to know. I left this one as-is because it's also storing the maxdiff so we can get an idea of how much the volume fractions differ from the baselines.